### ISSUE TITLE
Folder structure for data sources in Query manager

### ISSUE THREAD

**Issue by nechalmaggon** [ToolJet/ToolJet#9343]
### Description:
Data sources and their corresponding queries are currently scattered and lack a systematic organization. This feature proposes a folder structure to improve data management and accessibility.

Reference Figma design file for detailed folder structure

### Note:
**This item is in solutioning stage right now.**

**Duplicate item for the same feature:**
Folders in query manager #7140

**Comment by shubh22** [comment]
Duplicate.
